docs/generators/MSSqlDDL.md

```diff
@@ -0,0 +1,494 @@
+|UID                           |UNIQUEIDENTIFIER               |                             |UID           |UNIQUEIDENTIFIER     |
+|undefined SI type             |unkown                         |                             |*             |unknown              |
+
+The `hasSequenceReturning` variable is `False` by defualt and is set to `True` if the standard INSERT proc has the RETURNING keyword:
+```
+PROC INSERT RETURNING
+```
+
+# Automatic Adding of Fields
+## TmStamp
+The TmStamp field can automatically be added to the table by specifying the `--flag "add timestamp"` flag:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "add timestamp"
+```
+Example output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TIMESTAMP
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+```
+
+## UserId and TmStamp
+UserId's and TmStamp fields can automatically be added to the table by specifing `--flag "internal stamps"` to the generator.
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "internal stamps"
+```
+Example output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, UpdateWhen  DATETIME DEFAULT CURRENT_TIMESTAMP NULL
+, UpdateByWho CHAR(8)  DEFAULT USER NULL 
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+```
+
+# Triggers
+## Insert trigger
+An insert trigger can be used to automatically set `ID`, `TIMESTAMP` and `USERSTAMP` fields. In order for this to generate in the DDL you need to specify the Primary key of type `SEQUENCE` or `BIGSEQUENCE` and pass the `--flag "use insert trigger"` to the generator:
+
+Example si file:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+    TMSTAMP             TIMESTAMP
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "use insert trigger"
+```
+
+This will generate the DDL:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TMSTAMP DATETIME NOT NULL
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+
+IF OBJECT_ID('ExampleSchema.ExampleInsertTrigger','TR') IS NOT NULL
+    DROP TRIGGER ExampleSchema.ExampleInsertTrigger
+GO
+
+CREATE TRIGGER ExampleSchema.ExampleInsertTrigger ON ExampleSchema.Example FOR INSERT AS
+UPDATE ExampleSchema.Example SET ID=ID+0
+WHERE ID=(SELECT MAX(ID) FROM ExampleSchema.Example)
+UPDATE ExampleSchema.Example
+SET
```

Comment:
ID+0 doesn't seem right. Also, is this new functionality, or old, existing, undocumented functionality?

Reply:
OK, then my question is, +0 seems to be wrong. Surely it must be +1?

Reply:
Indeed. This does seem like a bug. The output updates the `ID` field to `Id+0` and then updates it again to `(SELECT MAX(ID) FROM ExampleSchema.Example)+1`. The first update seems to not do anything? Shall I create a new Issue?

Reply:
Yes, I think this entire generator may need an overhaul,